tests/outputs/TestDyd.cpp

```diff
@@ -316,3 +316,56 @@ TEST(Dyd, writeLoad) {
 
   dfl::test::checkFilesEqual(outputPath.generic_string(), reference.generic_string());
 }
+
+TEST(Dyd, writeVRRemote) {
+  using dfl::algo::HVDCDefinition;
+
+  std::string basename = "TestDydVRRemote";
+  std::string filename = basename + ".dyd";
+  boost::filesystem::path outputPath(outputPathResults);
+  outputPath.append(basename);
+
+  dfl::inputs::DynamicDataBaseManager manager("", "");
+
+  if (!boost::filesystem::exists(outputPath)) {
+    boost::filesystem::create_directories(outputPath);
+  }
+
+  std::shared_ptr<dfl::inputs::VoltageLevel> vl = std::make_shared<dfl::inputs::VoltageLevel>("VL");
+  std::shared_ptr<dfl::inputs::Node> node = dfl::inputs::Node::build("Slack", vl, 100., {});
+
+  auto hvdcLineLCC = HVDCDefinition("HVDCLCCLine", dfl::inputs::HvdcLine::ConverterType::LCC, "LCCStation1", "BUS_3", false, "LCCStation2",
+                                    "BUS_1", false, HVDCDefinition::Position::FIRST_IN_MAIN_COMPONENT, HVDCDefinition::HVDCModel::HvdcPTanPhiDangling,
+                                    {}, 0., boost::none, boost::none, boost::none, boost::none, false, 320, 322, 0.125, {0.01, 0.01});
+  auto hvdcLineVSC = HVDCDefinition("HVDCVSCLine", dfl::inputs::HvdcLine::ConverterType::VSC, "VSCStation1", "BUS_1", true, "VSCStation2",
+                                    "BUS_3", false, HVDCDefinition::Position::SECOND_IN_MAIN_COMPONENT, HVDCDefinition::HVDCModel::HvdcPVDangling, {},
+                                    0., boost::none, boost::none, boost::none, boost::none, false, 320, 322, 0.125, {0.01, 0.01});
+  HVDCLineDefinitions::HvdcLineMap hvdcLines = {std::make_pair(hvdcLineVSC.id, hvdcLineVSC), std::make_pair(hvdcLineLCC.id, hvdcLineLCC)};
+  HVDCLineDefinitions::BusVSCMap vscIds = {std::make_pair("BUS_1", "VSCStation1"), std::make_pair("BUS_3", "VSCStation2")};
+  HVDCLineDefinitions hvdcDefs{hvdcLines, vscIds};
+
+  const std::string bus1 = "BUS_1";
+  const std::string bus2 = "BUS_2";
+  GeneratorDefinitionAlgorithm::BusGenMap busesRegulatedBySeveralGenerators = {{bus1, "G1"}, {bus2, "G4"}};
+  DynamicModelDefinitions noModels;
+
+  outputPath.append(filename);
+  dfl::outputs::Dyd dydWriter(dfl::outputs::Dyd::DydDefinition(basename,
+                                                                outputPath.generic_string(),
+                                                                {},
+                                                                {},
+                                                                node,
+                                                                hvdcDefs,
+                                                                busesRegulatedBySeveralGenerators,
+                                                                manager,
+                                                                noModels,
+                                                                {}));
+
```

Comment:
I don't see any unit test where there are both hvdc and generators regulated by a VRRemote?

Reply:
In this unit test :

dydWriter.def_.dydVRRemote_->hvdcDefinitions_.vscBusVSCDefinitionsMap contains BUS_1 and BUS_3

dydWriter.def_.dydVRRemote_->busesRegulatedBySeveralGenerators_ contains BUS_1 and BUS_2

The unit test checks that only one Model_Signal_NQ is created for BUS_1 even if it's both regulated by a HVDC and a generator